bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/StreamingBigtableResultScanner.java

```diff
@@ -124,10 +125,12 @@ public void addResult(ReadRowsResponse response) {
   }
 
   public void setError(Throwable error) {
+    reservedChannel.returnToPool();
```

Comment:
The way we are reserving and returning channel is kind of scary. If we make a mistake somewhere and not returning channels, that will result in memory leak and keeping too many channels open.